modules/foreman_organization.py

```diff
@@ -57,44 +68,39 @@
     password: "changeme"
     server_url: "https://foreman.example.com"
     name: "My Cool New Organization"
+    state: present
 '''
 
 RETURN = '''# '''
 
 try:
-    from nailgun import entities
-    from nailgun.config import ServerConfig
+    from ansible.module_utils.ansible_nailgun_cement import (
+        create_server,
+        ping_server,
+        handle_no_nailgun,
+        find_entities,
+        naildown_entity_state,
+    )
+    from nailgun.entities import Organization
+
     HAS_NAILGUN_PACKAGE = True
 except:
     HAS_NAILGUN_PACKAGE = False
 
 
-class NailGun(object):
-    def __init__(self, server, entities, module):
-        self._server = server
-        self._entities = entities
-        self._module = module
-
-    def organization(self, name):
-        org = self._entities.Organization(self._server, name=name)
-        updated = False
-
-        response = org.search(set(), {'search': 'name="{}"'.format(name)})
-        if len(response) == 1:
-            org = response[0]
-        else:
-            org = None
+from ansible.module_utils.basic import AnsibleModule
 
-        if org and org.name != name:
-            org = self._entities.Organization(self._server, name=name, id=org.id)
-            org.update()
-            updated = True
-        elif not org:
-            org = self._entities.Organization(self._server, name=name)
-            org.create()
-            updated = True
 
-        return updated
+def sanitize_organization_dict(organization_dict):
+    # This is the only true source for names (and conversions thereof)
+    name_map = {
+        'name': 'name',
```

Comment:
I don't understand why this function is necessary if the `name_map` just includes `'name': 'name'`

Reply:
1st for consistency...
2nd, this function ensures, that no parameters not mentioned here are left in the dict (hence sanitize)
3rd, if in the future there were to be more parameters, it would be easy to add them.
4th, well consistency ;)

Reply:
Okay, makes sense. Silly question, could this be a part of cement in the future (not necessarily for this PR)?

Reply:
Yes, sounds good. I will look into it.
In the end we will need a written guide line `howto write modules with cement`.